src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final MachineProducer<Bulldozer> bulldozerMachineProducer;
+    private final MachineProducer<Excavator> excavatorMachineProducer;
+    private final MachineProducer<Truck> truckMachineProducer;
+
+    public MachineServiceImpl() {
+        bulldozerMachineProducer = new BulldozerProducer();
+        excavatorMachineProducer = new ExcavatorProducer();
+        truckMachineProducer = new TruckProducer();
+    }
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            return (List<T>) new ArrayList<>(bulldozerMachineProducer.get());
+        } else if (type == Excavator.class) {
+            return (List<T>) new ArrayList<>(excavatorMachineProducer.get());
+        } else if (type == Truck.class) {
+            return (List<T>) new ArrayList<>(truckMachineProducer.get());
+        }
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void startWorking(List<? extends T> machines) {
```

Comment:
you can simplify field params like 
```
void startWorking(List<T> machines)
```

Because T already represents the parametrized type of Machine.

Reply:
But it doesnt work, because program will ask for type "Machine" and types inheritors won`t compile, so i wrote following PECS rules: "List<? extend Machine>", because inheritors are producers for method. Is it ok for home-work?

Reply:
You should not ask questions here, mentor cant see them here, write to jv_chat